src/server/svr_jobfunc.c

```diff
@@ -3460,9 +3454,28 @@ eval_resvState(resc_resv *presv, enum resvState_discrim s, int relVal, int *psta
 	} else if (s == RESVSTATE_req_resvSub) {
 		*pstate = RESV_UNCONFIRMED;
 		*psub = RESV_UNCONFIRMED;
-	} else if (s == RESVSTATE_alter_failed)
-		/* backup only the state, as substate was not modified. */
-		*pstate = presv->ri_alter.ra_state;
+	} else if (s == RESVSTATE_alter_failed) {
+		if (presv->ri_alter.ra_state) {
+			*pstate = presv->ri_alter.ra_state;
+		} else if (*psub == RESV_IN_CONFLICT) {
+			if (is_running) {
+				*pstate = RESV_RUNNING;
+			} else {
+				*pstate = RESV_DEGRADED;
+			}
+		} else if (*psub == RESV_DEGRADED) {
+			*pstate = RESV_DEGRADED;
```

Comment:
If substate is degraded and is_running, pstate needs to be running.  Collapse this condition into the above condition.